<!doctype html>
<html>
<head>
<meta charset='UTF-8'><meta name='viewport' content='width=device-width initial-scale=1'>
<title>STAR</title><style type='text/css'>html {overflow-x: initial !important;}:root { --bg-color:#ffffff; --text-color:#333333; --select-text-bg-color:#B5D6FC; --select-text-font-color:auto; --monospace:"Lucida Console",Consolas,"Courier",monospace; }
html { font-size: 14px; background-color: var(--bg-color); color: var(--text-color); font-family: "Helvetica Neue", Helvetica, Arial, sans-serif; -webkit-font-smoothing: antialiased; }
body { margin: 0px; padding: 0px; height: auto; bottom: 0px; top: 0px; left: 0px; right: 0px; font-size: 1rem; line-height: 1.42857; overflow-x: hidden; background: inherit; tab-size: 4; }
iframe { margin: auto; }
a.url { word-break: break-all; }
a:active, a:hover { outline: 0px; }
.in-text-selection, ::selection { text-shadow: none; background: var(--select-text-bg-color); color: var(--select-text-font-color); }
#write { margin: 0px auto; height: auto; width: inherit; word-break: normal; overflow-wrap: break-word; position: relative; white-space: normal; overflow-x: visible; padding-top: 40px; }
#write.first-line-indent p { text-indent: 2em; }
#write.first-line-indent li p, #write.first-line-indent p * { text-indent: 0px; }
#write.first-line-indent li { margin-left: 2em; }
.for-image #write { padding-left: 8px; padding-right: 8px; }
body.typora-export { padding-left: 30px; padding-right: 30px; }
.typora-export .footnote-line, .typora-export li, .typora-export p { white-space: pre-wrap; }
@media screen and (max-width: 500px) {
  body.typora-export { padding-left: 0px; padding-right: 0px; }
  #write { padding-left: 20px; padding-right: 20px; }
  .CodeMirror-sizer { margin-left: 0px !important; }
  .CodeMirror-gutters { display: none !important; }
}
#write li > figure:last-child { margin-bottom: 0.5rem; }
#write ol, #write ul { position: relative; }
img { max-width: 100%; vertical-align: middle; }
button, input, select, textarea { color: inherit; font: inherit; }
input[type="checkbox"], input[type="radio"] { line-height: normal; padding: 0px; }
*, ::after, ::before { box-sizing: border-box; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p, #write pre { width: inherit; }
#write h1, #write h2, #write h3, #write h4, #write h5, #write h6, #write p { position: relative; }
p { line-height: inherit; }
h1, h2, h3, h4, h5, h6 { break-after: avoid-page; break-inside: avoid; orphans: 2; }
p { orphans: 4; }
h1 { font-size: 2rem; }
h2 { font-size: 1.8rem; }
h3 { font-size: 1.6rem; }
h4 { font-size: 1.4rem; }
h5 { font-size: 1.2rem; }
h6 { font-size: 1rem; }
.md-math-block, .md-rawblock, h1, h2, h3, h4, h5, h6, p { margin-top: 1rem; margin-bottom: 1rem; }
.hidden { display: none; }
.md-blockmeta { color: rgb(204, 204, 204); font-weight: 700; font-style: italic; }
a { cursor: pointer; }
sup.md-footnote { padding: 2px 4px; background-color: rgba(238, 238, 238, 0.7); color: rgb(85, 85, 85); border-radius: 4px; cursor: pointer; }
sup.md-footnote a, sup.md-footnote a:hover { color: inherit; text-transform: inherit; text-decoration: inherit; }
#write input[type="checkbox"] { cursor: pointer; width: inherit; height: inherit; }
figure { overflow-x: auto; margin: 1.2em 0px; max-width: calc(100% + 16px); padding: 0px; }
figure > table { margin: 0px !important; }
tr { break-inside: avoid; break-after: auto; }
thead { display: table-header-group; }
table { border-collapse: collapse; border-spacing: 0px; width: 100%; overflow: auto; break-inside: auto; text-align: left; }
table.md-table td { min-width: 32px; }
.CodeMirror-gutters { border-right: 0px; background-color: inherit; }
.CodeMirror-linenumber { user-select: none; }
.CodeMirror { text-align: left; }
.CodeMirror-placeholder { opacity: 0.3; }
.CodeMirror pre { padding: 0px 4px; }
.CodeMirror-lines { padding: 0px; }
div.hr:focus { cursor: none; }
#write pre { white-space: pre-wrap; }
#write.fences-no-line-wrapping pre { white-space: pre; }
#write pre.ty-contain-cm { white-space: normal; }
.CodeMirror-gutters { margin-right: 4px; }
.md-fences { font-size: 0.9rem; display: block; break-inside: avoid; text-align: left; overflow: visible; white-space: pre; background: inherit; position: relative !important; }
.md-diagram-panel { width: 100%; margin-top: 10px; text-align: center; padding-top: 0px; padding-bottom: 8px; overflow-x: auto; }
#write .md-fences.mock-cm { white-space: pre-wrap; }
.md-fences.md-fences-with-lineno { padding-left: 0px; }
#write.fences-no-line-wrapping .md-fences.mock-cm { white-space: pre; overflow-x: auto; }
.md-fences.mock-cm.md-fences-with-lineno { padding-left: 8px; }
.CodeMirror-line, twitterwidget { break-inside: avoid; }
.footnotes { opacity: 0.8; font-size: 0.9rem; margin-top: 1em; margin-bottom: 1em; }
.footnotes + .footnotes { margin-top: 0px; }
.md-reset { margin: 0px; padding: 0px; border: 0px; outline: 0px; vertical-align: top; background: 0px 0px; text-decoration: none; text-shadow: none; float: none; position: static; width: auto; height: auto; white-space: nowrap; cursor: inherit; -webkit-tap-highlight-color: transparent; line-height: normal; font-weight: 400; text-align: left; box-sizing: content-box; direction: ltr; }
li div { padding-top: 0px; }
blockquote { margin: 1rem 0px; }
li .mathjax-block, li p { margin: 0.5rem 0px; }
li { margin: 0px; position: relative; }
blockquote > :last-child { margin-bottom: 0px; }
blockquote > :first-child, li > :first-child { margin-top: 0px; }
.footnotes-area { color: rgb(136, 136, 136); margin-top: 0.714rem; padding-bottom: 0.143rem; white-space: normal; }
#write .footnote-line { white-space: pre-wrap; }
@media print {
  body, html { border: 1px solid transparent; height: 99%; break-after: avoid; break-before: avoid; }
  #write { margin-top: 0px; padding-top: 0px; border-color: transparent !important; }
  .typora-export * { -webkit-print-color-adjust: exact; }
  html.blink-to-pdf { font-size: 13px; }
  .typora-export #write { padding-left: 32px; padding-right: 32px; padding-bottom: 0px; break-after: avoid; }
  .typora-export #write::after { height: 0px; }
}
.footnote-line { margin-top: 0.714em; font-size: 0.7em; }
a img, img a { cursor: pointer; }
pre.md-meta-block { font-size: 0.8rem; min-height: 0.8rem; white-space: pre-wrap; background: rgb(204, 204, 204); display: block; overflow-x: hidden; }
p > .md-image:only-child:not(.md-img-error) img, p > img:only-child { display: block; margin: auto; }
p > .md-image:only-child { display: inline-block; width: 100%; }
#write .MathJax_Display { margin: 0.8em 0px 0px; }
.md-math-block { width: 100%; }
.md-math-block:not(:empty)::after { display: none; }
[contenteditable="true"]:active, [contenteditable="true"]:focus { outline: 0px; box-shadow: none; }
.md-task-list-item { position: relative; list-style-type: none; }
.task-list-item.md-task-list-item { padding-left: 0px; }
.md-task-list-item > input { position: absolute; top: 0px; left: 0px; margin-left: -1.2em; margin-top: calc(1em - 10px); border: none; }
.math { font-size: 1rem; }
.md-toc { min-height: 3.58rem; position: relative; font-size: 0.9rem; border-radius: 10px; }
.md-toc-content { position: relative; margin-left: 0px; }
.md-toc-content::after, .md-toc::after { display: none; }
.md-toc-item { display: block; color: rgb(65, 131, 196); }
.md-toc-item a { text-decoration: none; }
.md-toc-inner:hover { text-decoration: underline; }
.md-toc-inner { display: inline-block; cursor: pointer; }
.md-toc-h1 .md-toc-inner { margin-left: 0px; font-weight: 700; }
.md-toc-h2 .md-toc-inner { margin-left: 2em; }
.md-toc-h3 .md-toc-inner { margin-left: 4em; }
.md-toc-h4 .md-toc-inner { margin-left: 6em; }
.md-toc-h5 .md-toc-inner { margin-left: 8em; }
.md-toc-h6 .md-toc-inner { margin-left: 10em; }
@media screen and (max-width: 48em) {
  .md-toc-h3 .md-toc-inner { margin-left: 3.5em; }
  .md-toc-h4 .md-toc-inner { margin-left: 5em; }
  .md-toc-h5 .md-toc-inner { margin-left: 6.5em; }
  .md-toc-h6 .md-toc-inner { margin-left: 8em; }
}
a.md-toc-inner { font-size: inherit; font-style: inherit; font-weight: inherit; line-height: inherit; }
.footnote-line a:not(.reversefootnote) { color: inherit; }
.md-attr { display: none; }
.md-fn-count::after { content: "."; }
code, pre, samp, tt { font-family: var(--monospace); }
kbd { margin: 0px 0.1em; padding: 0.1em 0.6em; font-size: 0.8em; color: rgb(36, 39, 41); background: rgb(255, 255, 255); border: 1px solid rgb(173, 179, 185); border-radius: 3px; box-shadow: rgba(12, 13, 14, 0.2) 0px 1px 0px, rgb(255, 255, 255) 0px 0px 0px 2px inset; white-space: nowrap; vertical-align: middle; }
.md-comment { color: rgb(162, 127, 3); opacity: 0.8; font-family: var(--monospace); }
code { text-align: left; vertical-align: initial; }
a.md-print-anchor { white-space: pre !important; border-width: initial !important; border-style: none !important; border-color: initial !important; display: inline-block !important; position: absolute !important; width: 1px !important; right: 0px !important; outline: 0px !important; background: 0px 0px !important; text-decoration: initial !important; text-shadow: initial !important; }
.md-inline-math .MathJax_SVG .noError { display: none !important; }
.html-for-mac .inline-math-svg .MathJax_SVG { vertical-align: 0.2px; }
.md-math-block .MathJax_SVG_Display { text-align: center; margin: 0px; position: relative; text-indent: 0px; max-width: none; max-height: none; min-height: 0px; min-width: 100%; width: auto; overflow-y: hidden; display: block !important; }
.MathJax_SVG_Display, .md-inline-math .MathJax_SVG_Display { width: auto; margin: inherit; display: inline-block !important; }
.MathJax_SVG .MJX-monospace { font-family: var(--monospace); }
.MathJax_SVG .MJX-sans-serif { font-family: sans-serif; }
.MathJax_SVG { display: inline; font-style: normal; font-weight: 400; line-height: normal; zoom: 90%; text-indent: 0px; text-align: left; text-transform: none; letter-spacing: normal; word-spacing: normal; overflow-wrap: normal; white-space: nowrap; float: none; direction: ltr; max-width: none; max-height: none; min-width: 0px; min-height: 0px; border: 0px; padding: 0px; margin: 0px; }
.MathJax_SVG * { transition: none 0s ease 0s; }
.MathJax_SVG_Display svg { vertical-align: middle !important; margin-bottom: 0px !important; margin-top: 0px !important; }
.os-windows.monocolor-emoji .md-emoji { font-family: "Segoe UI Symbol", sans-serif; }
.md-diagram-panel > svg { max-width: 100%; }
[lang="mermaid"] svg, [lang="flow"] svg { max-width: 100%; height: auto; }
[lang="mermaid"] .node text { font-size: 1rem; }
table tr th { border-bottom: 0px; }
video { max-width: 100%; display: block; margin: 0px auto; }
iframe { max-width: 100%; width: 100%; border: none; }
.highlight td, .highlight tr { border: 0px; }


.CodeMirror { height: auto; }
.CodeMirror.cm-s-inner { background: inherit; }
.CodeMirror-scroll { overflow: auto hidden; z-index: 3; }
.CodeMirror-gutter-filler, .CodeMirror-scrollbar-filler { background-color: rgb(255, 255, 255); }
.CodeMirror-gutters { border-right: 1px solid rgb(221, 221, 221); background: inherit; white-space: nowrap; }
.CodeMirror-linenumber { padding: 0px 3px 0px 5px; text-align: right; color: rgb(153, 153, 153); }
.cm-s-inner .cm-keyword { color: rgb(119, 0, 136); }
.cm-s-inner .cm-atom, .cm-s-inner.cm-atom { color: rgb(34, 17, 153); }
.cm-s-inner .cm-number { color: rgb(17, 102, 68); }
.cm-s-inner .cm-def { color: rgb(0, 0, 255); }
.cm-s-inner .cm-variable { color: rgb(0, 0, 0); }
.cm-s-inner .cm-variable-2 { color: rgb(0, 85, 170); }
.cm-s-inner .cm-variable-3 { color: rgb(0, 136, 85); }
.cm-s-inner .cm-string { color: rgb(170, 17, 17); }
.cm-s-inner .cm-property { color: rgb(0, 0, 0); }
.cm-s-inner .cm-operator { color: rgb(152, 26, 26); }
.cm-s-inner .cm-comment, .cm-s-inner.cm-comment { color: rgb(170, 85, 0); }
.cm-s-inner .cm-string-2 { color: rgb(255, 85, 0); }
.cm-s-inner .cm-meta { color: rgb(85, 85, 85); }
.cm-s-inner .cm-qualifier { color: rgb(85, 85, 85); }
.cm-s-inner .cm-builtin { color: rgb(51, 0, 170); }
.cm-s-inner .cm-bracket { color: rgb(153, 153, 119); }
.cm-s-inner .cm-tag { color: rgb(17, 119, 0); }
.cm-s-inner .cm-attribute { color: rgb(0, 0, 204); }
.cm-s-inner .cm-header, .cm-s-inner.cm-header { color: rgb(0, 0, 255); }
.cm-s-inner .cm-quote, .cm-s-inner.cm-quote { color: rgb(0, 153, 0); }
.cm-s-inner .cm-hr, .cm-s-inner.cm-hr { color: rgb(153, 153, 153); }
.cm-s-inner .cm-link, .cm-s-inner.cm-link { color: rgb(0, 0, 204); }
.cm-negative { color: rgb(221, 68, 68); }
.cm-positive { color: rgb(34, 153, 34); }
.cm-header, .cm-strong { font-weight: 700; }
.cm-del { text-decoration: line-through; }
.cm-em { font-style: italic; }
.cm-link { text-decoration: underline; }
.cm-error { color: red; }
.cm-invalidchar { color: red; }
.cm-constant { color: rgb(38, 139, 210); }
.cm-defined { color: rgb(181, 137, 0); }
div.CodeMirror span.CodeMirror-matchingbracket { color: rgb(0, 255, 0); }
div.CodeMirror span.CodeMirror-nonmatchingbracket { color: rgb(255, 34, 34); }
.cm-s-inner .CodeMirror-activeline-background { background: inherit; }
.CodeMirror { position: relative; overflow: hidden; }
.CodeMirror-scroll { height: 100%; outline: 0px; position: relative; box-sizing: content-box; background: inherit; }
.CodeMirror-sizer { position: relative; }
.CodeMirror-gutter-filler, .CodeMirror-hscrollbar, .CodeMirror-scrollbar-filler, .CodeMirror-vscrollbar { position: absolute; z-index: 6; display: none; }
.CodeMirror-vscrollbar { right: 0px; top: 0px; overflow: hidden; }
.CodeMirror-hscrollbar { bottom: 0px; left: 0px; overflow: hidden; }
.CodeMirror-scrollbar-filler { right: 0px; bottom: 0px; }
.CodeMirror-gutter-filler { left: 0px; bottom: 0px; }
.CodeMirror-gutters { position: absolute; left: 0px; top: 0px; padding-bottom: 30px; z-index: 3; }
.CodeMirror-gutter { white-space: normal; height: 100%; box-sizing: content-box; padding-bottom: 30px; margin-bottom: -32px; display: inline-block; }
.CodeMirror-gutter-wrapper { position: absolute; z-index: 4; background: 0px 0px !important; border: none !important; }
.CodeMirror-gutter-background { position: absolute; top: 0px; bottom: 0px; z-index: 4; }
.CodeMirror-gutter-elt { position: absolute; cursor: default; z-index: 4; }
.CodeMirror-lines { cursor: text; }
.CodeMirror pre { border-radius: 0px; border-width: 0px; background: 0px 0px; font-family: inherit; font-size: inherit; margin: 0px; white-space: pre; overflow-wrap: normal; color: inherit; z-index: 2; position: relative; overflow: visible; }
.CodeMirror-wrap pre { overflow-wrap: break-word; white-space: pre-wrap; word-break: normal; }
.CodeMirror-code pre { border-right: 30px solid transparent; width: fit-content; }
.CodeMirror-wrap .CodeMirror-code pre { border-right: none; width: auto; }
.CodeMirror-linebackground { position: absolute; left: 0px; right: 0px; top: 0px; bottom: 0px; z-index: 0; }
.CodeMirror-linewidget { position: relative; z-index: 2; overflow: auto; }
.CodeMirror-wrap .CodeMirror-scroll { overflow-x: hidden; }
.CodeMirror-measure { position: absolute; width: 100%; height: 0px; overflow: hidden; visibility: hidden; }
.CodeMirror-measure pre { position: static; }
.CodeMirror div.CodeMirror-cursor { position: absolute; visibility: hidden; border-right: none; width: 0px; }
.CodeMirror div.CodeMirror-cursor { visibility: hidden; }
.CodeMirror-focused div.CodeMirror-cursor { visibility: inherit; }
.cm-searching { background: rgba(255, 255, 0, 0.4); }
@media print {
  .CodeMirror div.CodeMirror-cursor { visibility: hidden; }
}


:root { --side-bar-bg-color: #fafafa; --control-text-color: #777; }
html { font-size: 16px; }
body { font-family: "Open Sans", "Clear Sans", "Helvetica Neue", Helvetica, Arial, sans-serif; color: rgb(51, 51, 51); line-height: 1.6; }
#write { max-width: 860px; margin: 0px auto; padding: 30px 30px 100px; }
#write > ul:first-child, #write > ol:first-child { margin-top: 30px; }
a { color: rgb(65, 131, 196); }
h1, h2, h3, h4, h5, h6 { position: relative; margin-top: 1rem; margin-bottom: 1rem; font-weight: bold; line-height: 1.4; cursor: text; }
h1:hover a.anchor, h2:hover a.anchor, h3:hover a.anchor, h4:hover a.anchor, h5:hover a.anchor, h6:hover a.anchor { text-decoration: none; }
h1 tt, h1 code { font-size: inherit; }
h2 tt, h2 code { font-size: inherit; }
h3 tt, h3 code { font-size: inherit; }
h4 tt, h4 code { font-size: inherit; }
h5 tt, h5 code { font-size: inherit; }
h6 tt, h6 code { font-size: inherit; }
h1 { padding-bottom: 0.3em; font-size: 2.25em; line-height: 1.2; border-bottom: 1px solid rgb(238, 238, 238); }
h2 { padding-bottom: 0.3em; font-size: 1.75em; line-height: 1.225; border-bottom: 1px solid rgb(238, 238, 238); }
h3 { font-size: 1.5em; line-height: 1.43; }
h4 { font-size: 1.25em; }
h5 { font-size: 1em; }
h6 { font-size: 1em; color: rgb(119, 119, 119); }
p, blockquote, ul, ol, dl, table { margin: 0.8em 0px; }
li > ol, li > ul { margin: 0px; }
hr { height: 2px; padding: 0px; margin: 16px 0px; background-color: rgb(231, 231, 231); border: 0px none; overflow: hidden; box-sizing: content-box; }
li p.first { display: inline-block; }
ul, ol { padding-left: 30px; }
ul:first-child, ol:first-child { margin-top: 0px; }
ul:last-child, ol:last-child { margin-bottom: 0px; }
blockquote { border-left: 4px solid rgb(223, 226, 229); padding: 0px 15px; color: rgb(119, 119, 119); }
blockquote blockquote { padding-right: 0px; }
table { padding: 0px; word-break: initial; }
table tr { border-top: 1px solid rgb(223, 226, 229); margin: 0px; padding: 0px; }
table tr:nth-child(2n), thead { background-color: rgb(248, 248, 248); }
table tr th { font-weight: bold; border-width: 1px 1px 0px; border-top-style: solid; border-right-style: solid; border-left-style: solid; border-top-color: rgb(223, 226, 229); border-right-color: rgb(223, 226, 229); border-left-color: rgb(223, 226, 229); border-image: initial; border-bottom-style: initial; border-bottom-color: initial; margin: 0px; padding: 6px 13px; }
table tr td { border: 1px solid rgb(223, 226, 229); margin: 0px; padding: 6px 13px; }
table tr th:first-child, table tr td:first-child { margin-top: 0px; }
table tr th:last-child, table tr td:last-child { margin-bottom: 0px; }
.CodeMirror-lines { padding-left: 4px; }
.code-tooltip { box-shadow: rgba(0, 28, 36, 0.3) 0px 1px 1px 0px; border-top: 1px solid rgb(238, 242, 242); }
.md-fences, code, tt { border: 1px solid rgb(231, 234, 237); background-color: rgb(248, 248, 248); border-radius: 3px; padding: 2px 4px 0px; font-size: 0.9em; }
code { background-color: rgb(243, 244, 244); padding: 0px 2px; }
.md-fences { margin-bottom: 15px; margin-top: 15px; padding-top: 8px; padding-bottom: 6px; }
.md-task-list-item > input { margin-left: -1.3em; }
@media print {
  html { font-size: 13px; }
  table, pre { break-inside: avoid; }
  pre { overflow-wrap: break-word; }
}
.md-fences { background-color: rgb(248, 248, 248); }
#write pre.md-meta-block { padding: 1rem; font-size: 85%; line-height: 1.45; background-color: rgb(247, 247, 247); border: 0px; border-radius: 3px; color: rgb(119, 119, 119); margin-top: 0px !important; }
.mathjax-block > .code-tooltip { bottom: 0.375rem; }
.md-mathjax-midline { background: rgb(250, 250, 250); }
#write > h3.md-focus::before { left: -1.5625rem; top: 0.375rem; }
#write > h4.md-focus::before { left: -1.5625rem; top: 0.285714rem; }
#write > h5.md-focus::before { left: -1.5625rem; top: 0.285714rem; }
#write > h6.md-focus::before { left: -1.5625rem; top: 0.285714rem; }
.md-image > .md-meta { border-radius: 3px; padding: 2px 0px 0px 4px; font-size: 0.9em; color: inherit; }
.md-tag { color: rgb(167, 167, 167); opacity: 1; }
.md-toc { margin-top: 20px; padding-bottom: 20px; }
.sidebar-tabs { border-bottom: none; }
#typora-quick-open { border: 1px solid rgb(221, 221, 221); background-color: rgb(248, 248, 248); }
#typora-quick-open-item { background-color: rgb(250, 250, 250); border-color: rgb(254, 254, 254) rgb(229, 229, 229) rgb(229, 229, 229) rgb(238, 238, 238); border-style: solid; border-width: 1px; }
.on-focus-mode blockquote { border-left-color: rgba(85, 85, 85, 0.12); }
header, .context-menu, .megamenu-content, footer { font-family: "Segoe UI", Arial, sans-serif; }
.file-node-content:hover .file-node-icon, .file-node-content:hover .file-node-open-state { visibility: visible; }
.mac-seamless-mode #typora-sidebar { background-color: var(--side-bar-bg-color); }
.md-lang { color: rgb(180, 101, 77); }
.html-for-mac .context-menu { --item-hover-bg-color: #E6F0FE; }
#md-notification .btn { border: 0px; }
.dropdown-menu .divider { border-color: rgb(229, 229, 229); }
.ty-preferences .window-content { background-color: rgb(250, 250, 250); }
.ty-preferences .nav-group-item.active { color: white; background: rgb(153, 153, 153); }

 .typora-export li, .typora-export p, .typora-export,  .footnote-line {white-space: normal;} 
</style>
</head>
<body class='typora-export os-windows' >
<div  id='write'  class = 'is-node'><h1><a name="star" class="md-header-anchor"></a><span>STAR</span></h1><div class='md-toc' mdtype='toc'><p class="md-toc-content"><span class="md-toc-item md-toc-h1" data-ref="n1707"><a class="md-toc-inner" style="cursor: pointer;" href="#star">STAR</a></span><span class="md-toc-item md-toc-h2" data-ref="n1713"><a class="md-toc-inner" style="cursor: pointer;" href="#1-建立基因组索引">1. 建立基因组索引</a></span><span class="md-toc-item md-toc-h3" data-ref="n1714"><a class="md-toc-inner" style="cursor: pointer;" href="#11-基本选项">1.1 基本选项</a></span><span class="md-toc-item md-toc-h3" data-ref="n1716"><a class="md-toc-inner" style="cursor: pointer;" href="#12-代码实例">1.2 代码实例</a></span><span class="md-toc-item md-toc-h2" data-ref="n1721"><a class="md-toc-inner" style="cursor: pointer;" href="#2-序列比对">2. 序列比对</a></span><span class="md-toc-item md-toc-h3" data-ref="n1722"><a class="md-toc-inner" style="cursor: pointer;" href="#21-基本参数">2.1 基本参数</a></span><span class="md-toc-item md-toc-h3" data-ref="n1724"><a class="md-toc-inner" style="cursor: pointer;" href="#22-额外参数">2.2 额外参数</a></span><span class="md-toc-item md-toc-h3" data-ref="n1726"><a class="md-toc-inner" style="cursor: pointer;" href="#23-代码实例">2.3 代码实例</a></span><span class="md-toc-item md-toc-h2" data-ref="n1728"><a class="md-toc-inner" style="cursor: pointer;" href="#3-输出文件">3 输出文件</a></span><span class="md-toc-item md-toc-h3" data-ref="n1730"><a class="md-toc-inner" style="cursor: pointer;" href="#31-日志文件">3.1 日志文件</a></span><span class="md-toc-item md-toc-h3" data-ref="n1738"><a class="md-toc-inner" style="cursor: pointer;" href="#32-sam-sequence-alignment-map">3.2 SAM (Sequence Alignment Map)</a></span><span class="md-toc-item md-toc-h2" data-ref="n2103"><a class="md-toc-inner" style="cursor: pointer;" href="#4-starsolo">4. STARsolo</a></span><span class="md-toc-item md-toc-h3" data-ref="n2132"><a class="md-toc-inner" style="cursor: pointer;" href="#41-基本参数">4.1 基本参数</a></span><span class="md-toc-item md-toc-h3" data-ref="n2128"><a class="md-toc-inner" style="cursor: pointer;" href="#42-代码实例">4.2 代码实例</a></span></p></div><p><span>	</span><span>STAR全称</span><strong><span>S</span></strong><span>pliced </span><strong><span>T</span></strong><span>rans </span><strong><span>A</span></strong><span>lignment to a </span><strong><span>R</span></strong><span>eference，是由冷泉港实验室的Alexander Dobin等人员开发的为了解决large (&gt;80 billon reads) ENCODE Tranome RNA-seq dataset的有参比对问题。</span></p><p><span>	</span><span>STAR将read比对到基因组，相比于bowtie2更快，可以定量基因表达，且可以用于间接变体的拼接。但其弊端是，比对时不会考虑碱基的质量，比对准确率不如bowtie2。因此，在使用STAR进行比对前，必须对fastq文件做好质控，要把低质量序列去除。</span></p><p><span>	</span><span>STAR命令行遵循如下格式，如果一个选项可以输入多个参数，每个参数用空格分割，有几个特殊选项的参数可以用逗号分隔。</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="shell"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">STAR <span class="cm-attribute">--option1-name</span> option1-value(s) <span class="cm-attribute">--option2-name</span> option2-value(s)</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre><h2><a name="1-建立基因组索引" class="md-header-anchor"></a><span>1. 建立基因组索引</span></h2><h3><a name="11-基本选项" class="md-header-anchor"></a><span>1.1 基本选项</span></h3><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="shell"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 用于普通比对</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--runThreadN</span> <span class="cm-number">30</span> 使用线程数</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--runMode</span> genomeGenerate 运行模式为产生索引</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--genomeDir</span> /path/to/genomeDir 指定生成的基因组索引的保存目录</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--genomeFastaFiles</span> /path/to/genome/fasta1 建库使用的基因组序列</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 用于剪接变体</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--sjdbGTFfile</span> ./human.gtf 基因组注释文件 (for splicing junction data base)</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--sjdbOverhang</span> ReadLength-1 (for splicing junction data base) 数值为reads长度-1</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 184px;"></div><div class="CodeMirror-gutters" style="display: none; height: 184px;"></div></div></div></pre><h3><a name="12-代码实例" class="md-header-anchor"></a><span>1.2 代码实例</span></h3><p><span>建索引时可以加入gtf文件，定量基因表达时就不需要再指定gtf文件了。</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="shell"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 有gtf的索引</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">STAR <span class="cm-attribute">--runThreadN</span> <span class="cm-number">30</span> <span class="cm-attribute">--runMode</span> genomeGenerate <span class="cm-attribute">--genomeDir</span> ./starhuman150 <span class="cm-attribute">--genomeFastaFiles</span> hg38.fa <span class="cm-attribute">--sjdbGTFfile</span> ./human.gtf <span class="cm-attribute">--sjdbOverhang</span> <span class="cm-number">149</span></span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 69px;"></div><div class="CodeMirror-gutters" style="display: none; height: 69px;"></div></div></div></pre><p><span>如果索引不包含GTF文件，在定量基因表达时可以指定gtf文件路径，但必须保证gtf与fa文件的来源数据库和版本号相同。</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="shell"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 无gtf的索引</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">STAR <span class="cm-attribute">--runThreadN</span> <span class="cm-number">30</span> <span class="cm-attribute">--runMode</span> genomeGenerate <span class="cm-attribute">--genomeDir</span> ./starhuman150 <span class="cm-attribute">--genomeFastaFiles</span> hg38.fa</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 69px;"></div><div class="CodeMirror-gutters" style="display: none; height: 69px;"></div></div></div></pre><h2><a name="2-序列比对" class="md-header-anchor"></a><span>2. 序列比对</span></h2><h3><a name="21-基本参数" class="md-header-anchor"></a><span>2.1 基本参数</span></h3><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded md-focus" lang="shell"><div class="CodeMirror cm-s-inner CodeMirror-wrap CodeMirror-focused" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 69px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 参数调整</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--runThreadN</span> <span class="cm-number">6</span> 使用线程数</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--readFilesCommand</span> zcat OR gunzip <span class="cm-attribute">-c</span> 若输入的数据为gz压缩的文件，用zcat或gunzip进行解压</span></pre><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 数据输入</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--genomeDir</span> /path/to/genomeDir 基因组索引的目录</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--readFilesIn</span> /path/to/read1 [/path/to/read2] 输入测序数据fastq文件</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 数据输出</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--outSAMtype</span> BAM SortedByCoordinate 将结果SAM文件排序并生成BAM，效果与samtools类似</span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--outFileNamePrefix</span> AA 自定义生成文件的前缀</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 253px;"></div><div class="CodeMirror-gutters" style="display: none; height: 253px;"></div></div></div></pre><h3><a name="22-额外参数" class="md-header-anchor"></a><span>2.2 额外参数</span></h3><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="shell" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 单独指定注释文件，而不用在构建的时候使用</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--sjdbGTFfile</span> /path/to/ann.gtf</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--sjdbFileChrStartEnd</span> /path/to/sj.tab</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># ENCODE参数</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 减少伪junction的几率</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--outFilterType</span> BySJout</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 最多允许一个reads被匹配到多少个地方</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--outFilterMultimapNmax</span> <span class="cm-number">20</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 在未有注释的junction区域，最低允许突出多少个bp的单链序列</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--alignSJoverhangMin</span> <span class="cm-number">8</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 在有注释的junction区域，最低允许突出多少个bp的单链序列</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--alignSJDBoverhangMin</span> <span class="cm-number">1</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 过滤掉每个paired read mismatch数目超过N的数据，999代表着忽略这个过滤</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--outFilterMismatchNmax</span> <span class="cm-number">999</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 相对paired read长度可以允许的mismatch数目，如果read长度为2×100，则最大错配碱基数为0.04×200=8</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--outFilterMismatchNoverReadLmax</span> <span class="cm-number">0</span>.04</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 最小的intro长度</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--alignIntronMin</span> <span class="cm-number">20</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 最大的intro长度</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--alignIntronMax</span> <span class="cm-number">1000000</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 两条reads在基因组上的最大间隔</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--alignMatesGapMax</span> <span class="cm-number">1000000</span> </span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 736px;"></div><div class="CodeMirror-gutters" style="display: none; height: 736px;"></div></div></div></pre><h3><a name="23-代码实例" class="md-header-anchor"></a><span>2.3 代码实例</span></h3><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="shell"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">STAR <span class="cm-attribute">--runThreadN</span> <span class="cm-number">6</span> <span class="cm-attribute">--outSAMtype</span> BAM SortedByCoordinate <span class="cm-attribute">--genomeDir</span> ./starhuman150 <span class="cm-attribute">--readFilesIn</span> AA_R1_trimmed.fastq AA_R2_trimmed.fastq--outFileNamePrefix AA</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 46px;"></div><div class="CodeMirror-gutters" style="display: none; height: 46px;"></div></div></div></pre><h2><a name="3-输出文件" class="md-header-anchor"></a><span>3 输出文件</span></h2><p><span>STAR会产生多个输出文件。所有的文件都有标准的名字，但是，我们可以通过</span><code>--outFileNmaePrefix /path/to/output/dir/prefix</code><span>修改输出文件的目录，默认情况下输出文件放在当前目录下</span><code>./</code><span>。</span></p><h3><a name="31-日志文件" class="md-header-anchor"></a><span>3.1 日志文件</span></h3><ul><li><strong><span>Log.out</span></strong><span> 程序执行过程中的细节，可用于排除故障和调试代码。</span></li><li><strong><span>Log.progress.out</span></strong><span> 程序进程，每分钟记录一次。</span></li><li><strong><span>Log.final.out</span></strong><span> 比对结果统计信息。</span></li></ul><h3><a name="32-sam-sequence-alignment-map" class="md-header-anchor"></a><span>3.2 SAM (Sequence Alignment Map)</span></h3><p><span>输出</span><strong><span>Aligned.out.sam</span></strong><span> 比对结果。</span></p><ul><li><p><span>SAM的属性</span></p><p><span>输出的SAM文件的属性(字段)可以用</span><code>--outSAMattributes A1 A2 A3...</code><span>来指定。</span></p><p><span>关于SAM文件的详细介绍：</span></p><ol start='' ><li><a href='http://bioinformatics.cvr.ac.uk/blog/tag/cigar-string/' target='_blank' class='url'>http://bioinformatics.cvr.ac.uk/blog/tag/cigar-string/</a></li><li><a href='https://en.wikipedia.org/wiki/SAM_(file_format)' target='_blank' class='url'>https://en.wikipedia.org/wiki/SAM_(file_format)</a></li><li><a href='https://davetang.org/muse/2018/06/06/10x-single-cell-bam-files/' target='_blank' class='url'>https://davetang.org/muse/2018/06/06/10x-single-cell-bam-files/</a></li></ol><p><span>简单来说，SAM文件包含两部分: 1. header section，2. alignment section。Header section每行都以</span><code>@</code><span>开头，记录了reference中每条染色体的名字和长度等信息。Alignment section在header section之后，记录了每条reads的比对信息。Alignment section包含11个基础字段和多个可选字段。</span></p><p><span>查看10x 产生的SAM文件的header</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="shell"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">samtools view <span class="cm-attribute">-H</span> cellsorted_possorted_genome_bam.bam</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 23px;"></div><div class="CodeMirror-gutters" style="display: none; height: 23px;"></div></div></div></pre><h4><a name="基础字段" class="md-header-anchor"></a><span>基础字段</span></h4><figure><table><thead><tr><th><span>line</span></th><th><span>field</span></th><th><span>meaning</span></th></tr></thead><tbody><tr><td><span>1</span></td><td><span>QNAME</span></td><td><span>The name of the read.</span></td></tr><tr><td><span>2</span></td><td><span>FLAG</span></td><td><span>A bitwise number describing information about the read such as mapped/unmapped read, pairing, etc.</span></td></tr><tr><td><span>3</span></td><td><span>RNAME</span></td><td><span>The reference sequence name. The value of this field is * for unmapped reads.</span></td></tr><tr><td><span>4</span></td><td><span>POS</span></td><td><span>The left most mapping position of the first matching base. It is set to zero for unmapped reads.</span></td></tr><tr><td><span>5</span></td><td><span>MAPQ</span></td><td><span>A number to indicate the mapping quality.</span></td></tr><tr><td><span>6</span></td><td><span>CIGAR</span></td><td><span>A string describing how the read aligns with the reference. It consists of one or more components. Each component comprises an operator and the number of bases which the operator applies to. Operators are: MIDNSHP=X.  The following table gives an overview of these operators.</span></td></tr><tr><td><span>7</span></td><td><span>MNAME</span></td><td><span>The name of the mate strand, if it is a paired read. ‘=’ means that the name is the same as RNAME.</span></td></tr><tr><td><span>8</span></td><td><span>MPOS</span></td><td><span>A number indicating the left most mapping position of the mate.</span></td></tr><tr><td><span>9</span></td><td><span>TLEN</span></td><td><span>A number set to 0 for single-segment. For paired-segments it is calculated as the number of bases from the left most mapped base to the right most mapped base.</span></td></tr><tr><td><span>10</span></td><td><span>SEQ</span></td><td><span>The read sequence as the reference. If not  ‘*’ then the length of the sequence must be equal the sum of lengths of M/I/S/=/X operations in the CIGAR string.</span></td></tr><tr><td><span>11</span></td><td><span>QUAL</span></td><td><span>The read quality, same as the quality string in a fastaq format.</span></td></tr></tbody></table></figure><h4><a name="可选字段" class="md-header-anchor"></a><span>可选字段</span></h4><figure><table><thead><tr><th style='text-align:center;' ><span>Tag</span></th><th style='text-align:center;' ><span>Type</span></th><th style='text-align:center;' ><span>hideDescription</span></th></tr></thead><tbody><tr><td style='text-align:center;' ><span>AM</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>The smallest template-independent mapping quality in the template</span></td></tr><tr><td style='text-align:center;' ><span>AS</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Alignment score generated by aligner</span></td></tr><tr><td style='text-align:center;' ><span>BC</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Barcode sequence identifying the sample</span></td></tr><tr><td style='text-align:center;' ><span>BQ</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Offset to base alignment quality (BAQ)</span></td></tr><tr><td style='text-align:center;' ><span>BZ</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Phred quality of the unique molecular barcode bases in the OX tag</span></td></tr><tr><td style='text-align:center;' ><span>CB</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Cell identifier</span></td></tr><tr><td style='text-align:center;' ><span>CC</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Reference name of the next hit</span></td></tr><tr><td style='text-align:center;' ><span>CG</span></td><td style='text-align:center;' ><span>B,I</span></td><td style='text-align:center;' ><span>BAM only: CIGAR in BAM&#39;s binary encoding if (and only if) it consists of &gt;65535 operators</span></td></tr><tr><td style='text-align:center;' ><span>CM</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Edit distance between the color sequence and the color reference (see also NM)</span></td></tr><tr><td style='text-align:center;' ><span>CO</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Free-text comments</span></td></tr><tr><td style='text-align:center;' ><span>CP</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Leftmost coordinate of the next hit</span></td></tr><tr><td style='text-align:center;' ><span>CQ</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Color read base qualities</span></td></tr><tr><td style='text-align:center;' ><span>CR</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Cellular barcode sequence bases (uncorrected)</span></td></tr><tr><td style='text-align:center;' ><span>CS</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Color read sequence</span></td></tr><tr><td style='text-align:center;' ><span>CT</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Complete read annotation tag, used for consensus annotation dummy features</span></td></tr><tr><td style='text-align:center;' ><span>CY</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Phred quality of the cellular barcode sequence in the CR tag</span></td></tr><tr><td style='text-align:center;' ><span>E2</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>The 2nd most likely base calls</span></td></tr><tr><td style='text-align:center;' ><span>FI</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>The index of segment in the template</span></td></tr><tr><td style='text-align:center;' ><span>FS</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Segment suffix</span></td></tr><tr><td style='text-align:center;' ><span>FZ</span></td><td style='text-align:center;' ><span>B,S</span></td><td style='text-align:center;' ><span>Flow signal intensities</span></td></tr><tr><td style='text-align:center;' ><span>GC</span></td><td style='text-align:center;' ><span>?</span></td><td style='text-align:center;' ><span>Reserved for backwards compatibility reasons</span></td></tr><tr><td style='text-align:center;' ><span>GQ</span></td><td style='text-align:center;' ><span>?</span></td><td style='text-align:center;' ><span>Reserved for backwards compatibility reasons</span></td></tr><tr><td style='text-align:center;' ><span>GS</span></td><td style='text-align:center;' ><span>?</span></td><td style='text-align:center;' ><span>Reserved for backwards compatibility reasons</span></td></tr><tr><td style='text-align:center;' ><span>H0</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Number of perfect hits</span></td></tr><tr><td style='text-align:center;' ><span>H1</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Number of 1-difference hits (see also NM)</span></td></tr><tr><td style='text-align:center;' ><span>H2</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Number of 2-difference hits</span></td></tr><tr><td style='text-align:center;' ><span>HI</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Query hit index</span></td></tr><tr><td style='text-align:center;' ><span>IH</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Query hit total count</span></td></tr><tr><td style='text-align:center;' ><span>LB</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Library</span></td></tr><tr><td style='text-align:center;' ><span>MC</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>CIGAR string for mate/next segment</span></td></tr><tr><td style='text-align:center;' ><span>MD</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>String for mismatching positions</span></td></tr><tr><td style='text-align:center;' ><span>MF</span></td><td style='text-align:center;' ><span>?</span></td><td style='text-align:center;' ><span>Reserved for backwards compatibility reasons</span></td></tr><tr><td style='text-align:center;' ><span>MI</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Molecular identifier; a string that uniquely identifies the molecule from which the record was derived</span></td></tr><tr><td style='text-align:center;' ><span>MQ</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Mapping quality of the mate/next segment</span></td></tr><tr><td style='text-align:center;' ><span>NH</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Number of reported alignments that contain the query in the current record</span></td></tr><tr><td style='text-align:center;' ><span>NM</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Edit distance to the reference</span></td></tr><tr><td style='text-align:center;' ><span>OA</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Original alignment</span></td></tr><tr><td style='text-align:center;' ><span>OC</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Original CIGAR (deprecated; use OA instead)</span></td></tr><tr><td style='text-align:center;' ><span>OP</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Original mapping position (deprecated; use OA instead)</span></td></tr><tr><td style='text-align:center;' ><span>OQ</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Original base quality</span></td></tr><tr><td style='text-align:center;' ><span>OX</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Original unique molecular barcode bases</span></td></tr><tr><td style='text-align:center;' ><span>PG</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Program</span></td></tr><tr><td style='text-align:center;' ><span>PQ</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Phred likelihood of the template</span></td></tr><tr><td style='text-align:center;' ><span>PT</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Read annotations for parts of the padded read sequence</span></td></tr><tr><td style='text-align:center;' ><span>PU</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Platform unit</span></td></tr><tr><td style='text-align:center;' ><span>Q2</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Phred quality of the mate/next segment sequence in the R2 tag</span></td></tr><tr><td style='text-align:center;' ><span>QT</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Phred quality of the sample barcode sequence in the BC tag</span></td></tr><tr><td style='text-align:center;' ><span>QX</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Quality score of the unique molecular identifier in the RX tag</span></td></tr><tr><td style='text-align:center;' ><span>R2</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Sequence of the mate/next segment in the template</span></td></tr><tr><td style='text-align:center;' ><span>RG</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Read group</span></td></tr><tr><td style='text-align:center;' ><span>RT</span></td><td style='text-align:center;' ><span>?</span></td><td style='text-align:center;' ><span>Reserved for backwards compatibility reasons</span></td></tr><tr><td style='text-align:center;' ><span>RX</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Sequence bases of the (possibly corrected) unique molecular identifier</span></td></tr><tr><td style='text-align:center;' ><span>S2</span></td><td style='text-align:center;' ><span>?</span></td><td style='text-align:center;' ><span>Reserved for backwards compatibility reasons</span></td></tr><tr><td style='text-align:center;' ><span>SA</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Other canonical alignments in a chimeric alignment</span></td></tr><tr><td style='text-align:center;' ><span>SM</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Template-independent mapping quality</span></td></tr><tr><td style='text-align:center;' ><span>SQ</span></td><td style='text-align:center;' ><span>?</span></td><td style='text-align:center;' ><span>Reserved for backwards compatibility reasons</span></td></tr><tr><td style='text-align:center;' ><span>TC</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>The number of segments in the template</span></td></tr><tr><td style='text-align:center;' ><span>U2</span></td><td style='text-align:center;' ><span>Z</span></td><td style='text-align:center;' ><span>Phred probability of the 2nd call being wrong conditional on the best being wrong</span></td></tr><tr><td style='text-align:center;' ><span>UQ</span></td><td style='text-align:center;' ><span>i</span></td><td style='text-align:center;' ><span>Phred likelihood of the segment, conditional on the mapping being correct</span></td></tr><tr><td style='text-align:center;' ><span>X?</span></td><td style='text-align:center;' ><span>?</span></td><td style='text-align:center;' ><span>Reserved for end users</span></td></tr><tr><td style='text-align:center;' ><span>Y?</span></td><td style='text-align:center;' ><span>?</span></td><td style='text-align:center;' ><span>Reserved for end users</span></td></tr><tr><td style='text-align:center;' ><span>Z?</span></td><td style='text-align:center;' ><span>?</span></td><td style='text-align:center;' ><span>Reserved for end users</span></td></tr></tbody></table></figure><p><span>可选字段的值必须指定数据类型。</span></p><figure><table><thead><tr><th><span>Type</span></th><th><span>meaning</span></th></tr></thead><tbody><tr><td><strong><span>A</span></strong></td><td><span>character</span></td></tr><tr><td><strong><span>B</span></strong></td><td><span>general array</span></td></tr><tr><td><strong><span>f</span></strong></td><td><span>real number</span></td></tr><tr><td><strong><span>H</span></strong></td><td><span>hexadecimal array</span></td></tr><tr><td><strong><span>i</span></strong></td><td><span>integer</span></td></tr><tr><td><strong><span>Z</span></strong></td><td><span>string</span></td></tr></tbody></table></figure><p><span>查看10x产生的SAM文件的alignment</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="shell"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation"><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">samtools view cellsorted_possorted_genome_bam.bam | head <span class="cm-attribute">-1</span> | <span class="cm-builtin">awk</span> <span class="cm-tab" role="presentation" cm-text="	">  </span><span class="cm-string">'{for (f=1;f&lt;=NF;f++) col[f] = col[f]$f} END {for (f=1;f&lt;=NF;f++) print col[f]}'</span> | nl</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 46px;"></div><div class="CodeMirror-gutters" style="display: none; height: 46px;"></div></div></div></pre></li></ul><ul><li><p><span>SAM的排序</span>
<span>STAR可以直接输出BAM文件，同时根据坐标对BAM进行排序。排序后的BAM应用于很多下游分析。</span></p><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="shell"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 不排序</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--outSAMtype</span> BAM Unsorted</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 输出排序后的BAM文件：Aligned.sortedByCoord.out.bam</span></span></pre><pre class="CodeMirror-line" role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--outSAMtype</span> BAM SortedByCoordinate</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 输出不排序和排序的BAM文件</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--outSAMtype</span> BAM Unsorted SortedByCoordinate</span></pre></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 138px;"></div><div class="CodeMirror-gutters" style="display: none; height: 138px;"></div></div></div></pre></li></ul><h2><a name="4-starsolo" class="md-header-anchor"></a><span>4. STARsolo</span></h2><p><span>单细胞RNA-seq的比对和定量方案。主要包括以下四个步骤：</span></p><ul><li><span>根据用户提供的</span><a href='https://kb.10xgenomics.com/hc/en-us/articles/115004506263-What-is-a-barcode-whitelist-'><span>barcode白名单</span></a><span>，对barcode序列进行纠错和区分。注意：所用的白名单必须对应10x测序的试剂盒版本（如V1, V2, V3等）</span></li><li><span>利用标准的STAR spliced read比对算法将reads比对到基因组</span></li><li><span>对reads纠错并计算Unique Molecular Identifiers (UMI)</span></li><li><span>定量每个细胞的基因表达</span></li></ul><h3><a name="41-基本参数" class="md-header-anchor"></a><span>4.1 基本参数</span></h3><pre spellcheck="false" class="md-fences md-end-block ty-contain-cm modeLoaded" lang="shell" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre><span>xxxxxxxxxx</span></pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 启动STAR solo算法</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--soloType</span> Droplet</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 指定白名单文件</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--soloCBwhitelist</span> <span class="cm-comment"># /path/to/cell/barcode/whitelist</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 输入文件</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--readFilesIn</span> cDNAfragmentSequence.fastq.gz CellBarcodeUMIsequence.fastq.gz</span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Cell barcode起始位置</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--soloCBstart</span> <span class="cm-number">1</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># Cell barcode长度</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--soloCBlen</span> <span class="cm-number">16</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># UMI起始位置</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--soloUMIstart</span> <span class="cm-number">17</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># UMI长度</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--soloUMIlen</span> <span class="cm-number">10</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># barcode所在read的长度</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--soloBarcodeReadLength</span> <span class="cm-number">1</span> <span class="cm-comment"># '1'=soloCBlen+soloUMIlen '0'=不定义</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment"># 特征</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-attribute">--soloFeatures</span> Gene <span class="cm-comment"># Gene: 转录本; # SJ：剪接区域; # GeneFull: 外显子+内含子</span></span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 414px;"></div><div class="CodeMirror-gutters" style="display: none; height: 414px;"></div></div></div></pre><h3><a name="42-代码实例" class="md-header-anchor"></a><span>4.2 代码实例</span></h3><p><span>qsub任务提交脚本</span></p><pre lang="shell" class="md-fences md-end-block ty-contain-cm modeLoaded" spellcheck="false" style="break-inside: unset;"><div class="CodeMirror cm-s-inner CodeMirror-wrap" lang="shell"><div style="overflow: hidden; position: relative; width: 3px; height: 0px; top: 0px; left: 8px;"><textarea autocorrect="off" autocapitalize="off" spellcheck="false" tabindex="0" style="position: absolute; bottom: -1em; padding: 0px; width: 1000px; height: 1em; outline: none;"></textarea></div><div class="CodeMirror-scrollbar-filler" cm-not-content="true"></div><div class="CodeMirror-gutter-filler" cm-not-content="true"></div><div class="CodeMirror-scroll" tabindex="-1"><div class="CodeMirror-sizer" style="margin-left: 0px; margin-bottom: 0px; border-right-width: 0px; padding-right: 0px; padding-bottom: 0px;"><div style="position: relative; top: 0px;"><div class="CodeMirror-lines" role="presentation"><div role="presentation" style="position: relative; outline: none;"><div class="CodeMirror-measure"><pre>x</pre></div><div class="CodeMirror-measure"></div><div style="position: relative; z-index: 1;"></div><div class="CodeMirror-code" role="presentation" style=""><div class="CodeMirror-activeline" style="position: relative;"><div class="CodeMirror-activeline-background CodeMirror-linebackground"></div><div class="CodeMirror-gutter-background CodeMirror-activeline-gutter" style="left: 0px; width: 0px;"></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#PBS -l walltime=1336:00:00</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#PBS -l nodes=1:ppn=10,mem=20gb</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#PBS -j oe</span></span></pre><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#PBS -q cv2</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#PBS -o ESC.log</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-comment">#PBS -V </span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-builtin">cd</span> <span class="cm-def">$PBS_O_WORKDIR</span></span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">opts</span><span class="cm-operator">=</span><span class="cm-string">'--runThreadN 10 --readFilesCommand zcat'</span></span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">whitelist</span><span class="cm-operator">=</span><span class="cm-string">'--soloCBwhitelist 737K-august-2016.txt'</span></span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">mods</span><span class="cm-operator">=</span><span class="cm-string">'--soloType Droplet --soloFeatures Gene --soloBarcodeReadLength 1'</span></span></pre></div><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">outputs</span><span class="cm-operator">=</span><span class="cm-string">'--outSAMtype BAM SortedByCoordinate --outSAMattributes NH HI AS nM RE CR CY UR UY UB RG'</span></span></pre><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">filename</span><span class="cm-operator">=</span><span class="cm-string">'--outFileNamePrefix ESC'</span></span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span class="cm-def">reference</span><span class="cm-operator">=</span><span class="cm-string">'--genomeDir /public/home/jphe/omicsdata/genome/mm10/star_index/star2.7.0.f/protein_te'</span></span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;"><span cm-text="">​</span></span></pre></div><div class="" style="position: relative;"><pre class=" CodeMirror-line " role="presentation"><span role="presentation" style="padding-right: 0.1px;">STAR <span class="cm-def">$opts</span> <span class="cm-def">$whitelist</span> <span class="cm-def">$mods</span> <span class="cm-def">$mods</span> <span class="cm-def">$outputs</span> <span class="cm-def">$filename</span> <span class="cm-def">$reference</span> <span class="cm-attribute">--readFilesIn</span> fqs/ESC.p2.fq.gz fqs/ESC.p1.fq.gz</span></pre></div></div></div></div></div></div><div style="position: absolute; height: 0px; width: 1px; border-bottom: 0px solid transparent; top: 437px;"></div><div class="CodeMirror-gutters" style="display: none; height: 437px;"></div></div></div></pre><p>&nbsp;</p></div>
</body>
</html>